zstd_bullk_test.go

```diff
@@ -0,0 +1,152 @@
+package zstd
```

Comment:
could you maybe test corner cases (see other tests for `zstd_test.go` and `zstd_stream_test.go` and `zstd_ctx_test.go`) ? Also I think there is some general cleanup you could do to make the code clearer (I pointed some below)